### PR TITLE
[feature] Add support for SEPA direct debit in store and purchase with Stripe

### PR DESCRIPTION
**Pull request by dbackowski**
This will add support for SEPA Direct Debit payments for Stripe gateway

### CODE REVIEW

**Comment by dbackowski** [comment]
Screenshot 2021-01-12 at 11 25 06

File: lib/active_merchant/billing/gateways/stripe.rb [hunk 1]
```diff
@@ -222,7 +230,14 @@ def store(payment, options = {})
...
           end
         else
-          commit(:post, 'customers', post.merge(params), options)
+          if params[:payment_method_id].present?
+            MultiResponse.run(:first) do |r|
+              r.process { commit(:post, 'customers', post.merge(params.except(:payment_method_id)), options) }
+              r.process { setup_intents(r.responses.last.authorization.split("|").first, params[:payment_method_id], options) }
+            end
+          else
+            commit(:post, 'customers', post.merge(params.except(:payment_method_id)), options)
```
[the remark this answers is not in the document]
Comment (dbackowski) [reply lib/active_merchant/billing/gateways/stripe.rb:239]: What can I say 🤦

File: lib/active_merchant/billing/gateways/stripe.rb [hunk 2]
```diff
@@ -338,7 +359,7 @@ def create_post_for_auth_or_purchase(money, payment, options)
         post
       end
 
-      def card_payment_method_for_customer(customer)
+      def card_payment_method_for_customer(customer, payment_type)
```
[the remark this answers is not in the document]
Comment (dbackowski) [reply lib/active_merchant/billing/gateways/stripe.rb:362]: Well, I must have missed that, probably in the first version I was using that param, but later I decided to create another method `sepa_debit_payment_method_for_customer`. I will remove this param, good catch 👍

File: lib/active_merchant/billing/gateways/stripe.rb [hunk 3]
```diff
@@ -365,6 +386,23 @@ def card_payment_method_for_customer(customer)
...
+      def sepa_debit_payment_method_for_customer(customer)
+        r = commit(:get, "payment_methods?customer=#{customer}&type=sepa_debit", nil, options)
+        raise r.message unless r.success?
+
+        payment_methods = r.params["data"]
+        return payment_methods[0]["id"] if payment_methods&.count == 1
+
+        r = commit(:get, "customers/#{customer}", nil, options)
+        # if customer has default payment method
+        default_payment_method = r.params.dig("invoice_settings", "default_payment_method")
```
[the remark this answers is not in the document]
Comment (dbackowski) [reply lib/active_merchant/billing/gateways/stripe.rb:398]: Yes we have a part of this logic in the `card_payment_method_for_customer` method, will try to move this shared part into another method

[the remark this answers is not in the document]
Comment (dbackowski) [reply lib/active_merchant/billing/gateways/stripe.rb:397]: Done 👍

**Review by himn1 — COMMENTED** [review]
looks good

**Review by himn1 — APPROVED** [review]
🚢

**Pull request merged**